Gruntfile.dashboard.js

```diff
@@ -0,0 +1,350 @@
+'use strict'
+
+// ========================================
+// A Gruntfile.js for chelonia-dashboard front-end
+// ========================================
+
+const path = require('path')
+const chalk = require('chalk')
+const crypto = require('crypto')
+const { readFile } = require('fs/promises')
+const { fork } = require('child_process')
+
+const applyPortShift = (env) => {
+  // TODO: implement automatic port selection when `PORT_SHIFT` is 'auto'.
+  const API_PORT = 8000 + Number.parseInt(env.PORT_SHIFT || '0')
+  const API_URL = 'http://127.0.0.1:' + API_PORT
+
+  if (Number.isNaN(API_PORT) || API_PORT < 8000 || API_PORT > 65535) {
+    throw new RangeError(`Invalid API_PORT value: ${API_PORT}.`)
+  }
+  return { ...env, API_PORT, API_URL }
+}
+
+Object.assign(process.env, applyPortShift(process.env))
+
+// paths
+const dashboardRootDir = path.resolve(__dirname, 'backend/dashboard')
+const resolvePathFromRoot = relPath => path.join(dashboardRootDir, relPath) // resolve a relative path against backend/dashboard folder
+const distDir = path.resolve('dist-dashboard')
+const distAssets = path.join(distDir, 'assets')
+const distCSS = path.join(distDir, 'assets/css')
+const distJS = path.join(distDir, 'assets/js')
+const mainSrc = resolvePathFromRoot('main.js')
+const mainScss = resolvePathFromRoot('assets/style/main.scss')
+const backendIndex = 'backend/index.js'
+
+module.exports = (grunt) => {
+  const isDevelopment = grunt.option('development') === true
+  const isProduction = grunt.option('production') === true
```

Comment:
Can we keep this consistent, so that they are set using `NODE_ENV` environment variable? It is standard practice to use `NODE_ENV`. Feel free to copy/paste how it was done in the original Gruntfile.